kie-wb-common-react/boxed-expression-editor/src/components/ContextExpression/ContextEntryInfoCell.tsx

```diff
@@ -32,24 +33,33 @@ export const ContextEntryInfoCell: React.FunctionComponent<ContextEntryInfoCellP
 }) => {
   const contextEntry = data[index];
 
-  const [entryInfo, setEntryInfo] = useState(contextEntry.entryInfo);
+  const entryInfo = useRef(contextEntry.entryInfo);
+  const entryExpression = useRef(contextEntry.entryExpression);
 
   useEffect(() => {
-    setEntryInfo(contextEntry.entryInfo);
+    entryInfo.current = contextEntry.entryInfo;
   }, [contextEntry.entryInfo]);
+  useEffect(() => {
+    entryExpression.current = contextEntry.entryExpression;
+  }, [contextEntry.entryExpression]);
 
   const onContextEntryUpdate = useCallback(
     (name, dataType) => {
-      onRowUpdate(index, { ...contextEntry, entryInfo: { name, dataType } });
+      const updatedExpression = { ...entryExpression.current };
+      if (contextEntry.nameAndDataTypeSynchronized && _.size(name) && _.size(dataType)) {
+        updatedExpression.name = name;
+        updatedExpression.dataType = dataType;
+      }
```

Comment:
probably sth. like this would be needed also for a function

Reply:
Is this related to https://github.com/karreiro/kogito-editors-java/pull/13#issuecomment-877000274 ?

Reply:
yes, as it is reported separately, closing this